akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -302,8 +302,8 @@ import scala.collection.JavaConverters._
           nextElementValue = elem
         }
       } else {
-        if (activeSubstreamsMap.size == maxSubstreams)
-          fail(new IllegalStateException(s"Cannot open substream for key '$key': too many substreams open"))
+        if (activeSubstreamsMap.size + closedSubstreams.size == maxSubstreams)
```

Comment:
I know you have had lengthy discussions about this, but let's clarify the reason for this specific change here (also).

While I agree that also small memory "leaks" are important it the closedSubstreams only contains the key and it's a somewhat unfair to count them equally to active substreams. Perhaps we should use some fuzzy multiplier for the closed, such as ` + closedSubstreams.size / 100` ?

Reply:
The check being made is against the max number of substreams - whether they are active or not. I think that my change is in line with the API doc and that what I’m doing here is just enforcing the existing spec. 

If there remains a concern then I’d suggest dropping the PR altogether.

Reply:
Any further thoughts on this @patriknw? Shall we close it, rather than have it linger? I feel that it is a valid PR to pursue, but I don't like having PRs hang around too long.

If we do close it then I propose that the API doc has a big fat warning that `groupBy` will grow memory in an unbounded fashion. :-)

Reply:
Thanks for the ping. I'm fine with having a stricter count as you propose here. Any objections @ktoso or others?

Reply:
The PR has been updated @patriknw - the tests should now pass. However, I'll admit to not spending enough time on grokking the test adjustment. It feels intuitive, but it should be checked.

Reply:
@patriknw Shall we close this? I'm concerned around the hesitation of including the check.